apps/server/src/modules/board/uc/media-board/media-available-line.uc.ts

```diff
@@ -38,8 +42,22 @@ export class MediaAvailableLineUc {
 			schoolExternalToolsForAvailableMediaLine
 		);
 
-		const mediaAvailableLine: MediaAvailableLine =
-			this.mediaAvailableLineService.createMediaAvailableLine(matchedTools);
+		let matchedAndlicensedTools: [ExternalTool, SchoolExternalTool][] = [];
+		if (this.licenseConfigService.get('FEATURE_SCHULCONNEX_MEDIA_LICENSE_ENABLED')) {
+			const mediaUserLicenses: MediaUserLicense[] = await this.userLicenseService.getMediaUserLicensesForUser(userId);
+
+			matchedAndlicensedTools = matchedTools.filter((tool: [ExternalTool, SchoolExternalTool]): boolean => {
+				const externalTool: ExternalTool = tool[0];
+				if (externalTool?.medium?.mediumId) {
+					return this.userLicenseService.haslicenseForExternalTool(externalTool.medium.mediumId, mediaUserLicenses);
+				}
+				return true;
+			});
+		}
+
+		const mediaAvailableLine: MediaAvailableLine = this.mediaAvailableLineService.createMediaAvailableLine(
+			matchedAndlicensedTools.length ? matchedAndlicensedTools : matchedTools
```

Comment:
why two different results? I would prefer working on only one and filter tools which has no licenses but requires one